queries/dune_v2/period_slippage.sql

```diff
@@ -84,42 +85,42 @@ other_transfers as (
                 and evt_tx_hash = b.tx_hash
              inner join batchwise_traders bt
                 on evt_tx_hash = bt.tx_hash
-    where b.block_time between '{{StartTime}}' and '{{EndTime}}'
-      and '0x9008d19f58aabd9ed0d60971565aa8510560ab41' in (to, from)
-      and not array_contains(traders_in, from)
-      and not array_contains(traders_out, to)
-      and from not in ( -- ETH FLOW ORDERS ARE NOT AMM TRANSFERS!
+    where b.block_time between cast('{{StartTime}}' as timestamp) and cast('{{EndTime}}' as timestamp)
+      and 0x9008d19f58aabd9ed0d60971565aa8510560ab41 in (to, "from")
+      and not contains(traders_in, "from")
+      and not contains(traders_out, to)
+      and "from" not in ( -- ETH FLOW ORDERS ARE NOT AMM TRANSFERS!
           select distinct contract_address
           from cow_protocol_ethereum.CoWSwapEthFlow_evt_OrderPlacement
       )
-      and (t.evt_tx_hash = lower('{{TxHash}}') or '{{TxHash}}' = '0x')
-      and (solver_address = lower('{{SolverAddress}}') or '{{SolverAddress}}' = '0x')
-),
-eth_transfers as (
+      and (t.evt_tx_hash = from_hex('{{TxHash}}') or '{{TxHash}}' = '0x')
+      and (solver_address = from_hex('{{SolverAddress}}') or '{{SolverAddress}}' = '0x')
+)
+,eth_transfers as (
     select
         bt.tx_hash,
-        from as sender,
-        to as receiver,
-        '0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee' as token,
-        cast(value as decimal(38, 0)) as amount_wei,
+        "from" as sender,
+        to     as receiver,
+        0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee as token,
+        cast(value as int256) as amount_wei,
         case
-          when '0x9008d19f58aabd9ed0d60971565aa8510560ab41' = to
+          when 0x9008d19f58aabd9ed0d60971565aa8510560ab41 = to
           then 'AMM_IN'
           else 'AMM_OUT'
         end as transfer_type
     from batchwise_traders bt
     inner join ethereum.traces et
         on bt.block_number = et.block_number
         and bt.tx_hash = et.tx_hash
-        and cast(value as decimal(38, 0)) > 0
+        and value > cast(0 as uint256)
```

Comment:
because value is uint256 (and we are comparing value with this number). `uint` is (unsigned -- nonnegative) while `int` is both positive and negative. 

The other places are being cast to int because they can be negative, but in this place I am only concerned with value being non-zero. Specifically this is a subquery extracting non-trivial "ETH transfers from `ethereum.traces` table.

Reply:
Okay, thanks for the explanation.